core/src/main/java/com/amplifyframework/datastore/DataStoreQuerySnapshot.java

```diff
@@ -0,0 +1,30 @@
+package com.amplifyframework.datastore;
+
+import com.amplifyframework.core.model.Model;
+
+import java.util.List;
+
+public class DataStoreQuerySnapshot<T extends Model> {
+    private final List<T> items;
+    private final boolean isSynced;
+    // Why do we have items and itemChanges and which do we return on
+    private final List<DataStoreItemChange<T>> itemChanges;
+    public DataStoreQuerySnapshot(List<T> items, boolean isSynced, List<DataStoreItemChange<T>> itemChanges){
```

Comment:
For the list items, use `Immutable.of()` utility to create an unmodifiable copy of the data. Can apply that same trick around the PR. Existing pattern is shown in the code base.

Reply:
Thanks for pointing that out. Will apply it where ever applicable. :)